Query stuff

lukedirtwalker opened the pull request:
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/223%23issuecomment-159534890%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/223%23issuecomment-159535841%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/223%23issuecomment-159534890%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20think%20I%20fixed%20the%202%20things%20we%20agreed%20to%20fix%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-11-25T08%3A39%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3530769%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/lukedirtwalker%22%7D%7D%2C%20%7B%22body%22%3A%20%22Thanks.%22%2C%20%22created_at%22%3A%20%222015-11-25T08%3A44%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-General Comment'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/223#issuecomment-159534890'>General Comment</a></b>
- <a href='https://github.com/lukedirtwalker'><img border=0 src='https://avatars.githubusercontent.com/u/3530769?v=3' height=16 width=16'></a> I think I fixed the 2 things we agreed to fix :)
- <a href='https://github.com/dimitar-asenov'><img border=0 src='https://avatars.githubusercontent.com/u/1152976?v=3' height=16 width=16'></a> Thanks.

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/223?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/223?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/223'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

lukedirtwalker committed:
Fix profileSource.py script

lukedirtwalker committed:
Expose AstModification to python environment

lukedirtwalker committed:
Move PropertyDataConcept in detail namespace.

Templated structs inside a class is not the best idea, especially since
it prevents full specialization.

lukedirtwalker committed:
Delete empty Tuple constructor

The arguments are required so that we can initialize the tag.

lukedirtwalker committed:
Make it possible to show newlines in queryprompt showErrors()

We need this for script errors.

lukedirtwalker committed:
Remove the requirement to write NamedProperty in python scripts

lukedirtwalker committed:
Make it possible to build queries from strings in python

lukedirtwalker committed:
Use Query.executQuery for filterCasts.py

lukedirtwalker committed:
Change the default behavior of toParent query

By default add nodes instead of adding the relation. In all example we
always expect nodes and never the relation, thus this should be the
default behavior.

dimitar-asenov commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/223'>Refresh</a>.<a href='#crh-update-none'></a>

lukedirtwalker committed:
Make bases query work on input and globally as well

lukedirtwalker committed:
Make it possible to add callgraph globally

lukedirtwalker committed:
Remove wrapInStatement form AstModification

Scripts can create nodes with Node.createNewNode(...) so we don't such a
wrapping function.

lukedirtwalker committed:
Auto erase spaces after pipe symbol

Currently parsing "methods  |    color" would not detect color as a
command. So we just erase all the spaces before.

lukedirtwalker commented:
I think I fixed the 2 things we agreed to fix :)

dimitar-asenov commented:
Thanks.
